src/Kafker/Helpers/EmitPreserveTime.cs

```diff
@@ -0,0 +1,133 @@
+            var producedEvents = 0;
+
+            try
+            {
+                var getDictionary = await CreatePreserveTime(fileName);
+                var tasks = getDictionary.Select(x => SetTimer(x.Key, x.Value, cfg,cancellationToken)).ToArray();
+
+                startEvent = true;
+                Task.WaitAll(tasks);
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine(e);
+                throw;
+            }
+            finally
+            {
+                await _console.Out.WriteLineAsync($"\r\nProduced {producedEvents} events");
+            }
+
+            return await Task.FromResult(0).ConfigureAwait(false); // ok
+        }
+
+        private async Task<Dictionary<long, List<string>>> CreatePreserveTime(string fileName)
+        {
+            var allLines = await File.ReadAllLinesAsync(fileName);
+            var timeStampWithEvents = new List<Tuple<long, string>>();
+            var eventsToEmit = new Dictionary<long, List<string>>();
+            var initialSnapshot = allLines.Select(item => item.Split("|")).Select(pair => new Tuple<long, string>(long.Parse(pair[0].Replace("\"", "")), pair[1])).ToList();
+            var smallest = initialSnapshot.Select(x => x.Item1).Min();
+
+            foreach (var (timeStamp, itemToSend) in initialSnapshot)
+            {
+                timeStampWithEvents.Add(new Tuple<long, string>(timeStamp - smallest, itemToSend));
+            }
+
+            foreach (var (timeStamp, itemToSend) in timeStampWithEvents)
+            {
+                if (!eventsToEmit.ContainsKey(timeStamp))
+                {
+                    eventsToEmit.Add(timeStamp, new List<string>());
+                }
+
+                eventsToEmit[timeStamp].Add(itemToSend);
+            }
+
+            return eventsToEmit;
+        }
+
+        private static async Task SetTimer(long key, List<string> value, KafkaTopicConfiguration config,CancellationToken cancellationToken)
+        {
+            await Task.Yield();
+            var stopEvent = new AutoResetEvent(false);
+
+            var data = new PreserveTime
+            {
+                ItemsToSend = value,
+                ProducerConfig = config,
+                Waiter = stopEvent
+            };
+            while (!startEvent)
+            {
+                await Task.Delay(1, cancellationToken);
+                if (cancellationToken.IsCancellationRequested) return;
+            }
+
+            var ts = TimeSpan.FromTicks(key);
+            await Task.Delay(ts, cancellationToken);
+            if (!cancellationToken.IsCancellationRequested)
+            {
+                var timer = new Timer(EmitEventsOnTime, data, key, Timeout.Infinite);
+                stopEvent.WaitOne();
+            }
+        }
+
+        private static void EmitEventsOnTime(object state)
+        {
+            if (state != null)
```

Comment:
In this method you need to use unblocking synchronization of Tasks to send all events almost simultaneously.
Hint: 
```
var tasks  = new []{Task, task,....};
await Task.WhenAll(tasks);
```